lib/fencing/st_rhcs.c

```diff
@@ -138,23 +138,31 @@ stonith__rhcs_get_metadata(const char *agent, int timeout, xmlNode **metadata)
         return rc;
     }
 
-    stonith__action_result(action, &rc, &buffer, NULL);
-    stonith__destroy_action(action);
-    if (rc < 0) {
-        crm_warn("Metadata action for %s failed: %s " CRM_XS "rc=%d",
-                 agent, pcmk_strerror(rc), rc);
-        free(buffer);
-        return rc;
+    result = stonith__action_result(action);
+
+    if (result->execution_status != PCMK_EXEC_DONE) {
+        crm_warn("Could not execute metadata action for %s: %s",
+                 agent, pcmk_exec_status_str(result->execution_status));
+        stonith__destroy_action(action);
+        return pcmk_rc2legacy(stonith__result2rc(result));
```

Comment:
shouldn't the result be gone with the action?

Reply:
whoops, definitely